armonik-client/src/main/java/fr/aneo/armonik/client/ManagedChannelPool.java

```diff
@@ -267,47 +267,57 @@ public static ManagedChannelPool create(int maxSize, RetryPolicy retryPolicy, Su
   }
 
   /**
-   * Acquires a channel from the pool.
+   * Acquires a {@link ManagedChannel} from the pool.
    * <p>
-   * For bounded pools: If no channels are available and the pool has reached its maximum size,
-   * this method will block until a channel is released by another thread.
+   * Behavior differs depending on the pool mode:
+   * <ul>
+   *   <li><strong>Bounded pool:</strong> If all channels are in use and the pool has reached its maximum size,
+   *   this call blocks until a channel is released by another thread.</li>
+   *   <li><strong>Unbounded pool:</strong> Always creates a new channel if none are available. Never blocks.</li>
+   * </ul>
    * <p>
-   * For unbounded pools: Always creates a new channel if none are available. Never blocks.
+   * When acquiring a channel:
+   * <ul>
+   *   <li>The method polls the internal queue of available channels.</li>
+   *   <li>Each polled channel is validated via {@link #isChannelHealthy(ManagedChannel)}.</li>
+   *   <li>Unhealthy channels are automatically removed from the pool and shut down.</li>
+   *   <li>The first healthy channel encountered is returned for reuse.</li>
+   *   <li>If no healthy channel is found, a new one is created using the configured {@code channelFactory}.</li>
+   * </ul>
    *
-   * @return a managed gRPC channel ready for use
+   * @return a healthy {@link ManagedChannel} ready for use
    * @throws IllegalStateException if the pool has been shut down
-   * @throws InterruptedException  if interrupted while waiting for a channel (bounded pools only)
+   * @throws InterruptedException  if interrupted while waiting for a channel in a bounded pool
    */
   private ManagedChannel acquire() throws InterruptedException {
     ensureNotShutdown();
 
     if (!isUnbounded) {
       semaphore.acquire();
     }
-
     try {
-      ManagedChannel channel = availableChannels.poll();
-
-      if (channel != null) {
-        if (isChannelHealthy(channel)) {
+      ManagedChannel channel = null;
+      ManagedChannel candidate;
```

Comment:
You don't really need to distinguish between the candidate and the actual channel if you reset it back to null when it is unhealthy.